src/components/AdsSlider/AdsSlider.tsx

```diff
@@ -13,6 +13,9 @@ const AdsSlider: React.FC<AdsSliderProps> = ({ sort }) => {
   const adsData = useDefaultAdsList();
   const ads = adsData.data[sort];
 
+  const startIndex =
```

Comment:
No, this doesn't work. Only generate a random between the 0 and length..

You can do this using the modulo operator
Random % length